### PR TITLE
fix unprivileged mode on Linux

### PR DESCRIPTION
**Pull request by schweikert**

### CODE REVIEW

File: src/fping.c [hunk 1]
```diff
@@ -1104,11 +1104,11 @@ int main(int argc, char** argv)
         exit(num_noaddress ? 2 : 1);
     }
 
-    if (src_addr_set && socket4 >= 0) {
+    if (socket4 && (src_addr_set || socktype4 == SOCK_DGRAM)) {
```
Comment (sfan5) [inline src/fping.c:1107]: Shouldn't this still be `socket4 >= 0`? 0 is theoretically a valid fd while -1 is not

Comment (schweikert) [reply src/fping.c]: Yes, thanks for catching this.

File: src/fping.c [hunk 2]
```diff
@@ -1104,11 +1104,11 @@ int main(int argc, char** argv)
         exit(num_noaddress ? 2 : 1);
     }
 
-    if (src_addr_set && socket4 >= 0) {
+    if (socket4 >= 0 && (src_addr_set || socktype4 == SOCK_DGRAM)) {
         socket_set_src_addr_ipv4(socket4, &src_addr, (socktype4 == SOCK_DGRAM) ? &ident4 : NULL);
     }
 #ifdef IPV6
-    if (src_addr6_set && socket6 >= 0) {
+    if (socket6 >= 0 && (src_addr6_set || socktype4 == SOCK_DGRAM)) {
```
Comment (sfan5) [inline src/fping.c:1111]: One more: it says `socktype4` on this line, copy-paste mistake?

Comment (schweikert) [reply src/fping.c:1111]: Oh my... fixed as well. thanks!

**Pull request merged**